MDANSE/Src/MDANSE/Framework/Jobs/NeutronDynamicTotalStructureFactor.py

```diff
@@ -400,34 +400,53 @@ def finalize(self):
             bj = self.configuration["trajectory"]["instance"].get_atom_property(
                 pair[1], "b_coherent"
             )
+            sqrt_cij = sqrt(
+                nAtomsPerElement[pair[0]] * nAtomsPerElement[pair[1]] * norm_natoms**2
+            )
 
             if pair[0] == pair[1]:  # Add a factor 2 if the two elements are different
-                self._outputData[f"f(q,t)_coh_{pair_str}"] *= bi * bj * norm_natoms
-                self._outputData[f"s(q,f)_coh_{pair_str}"] *= bi * bj * norm_natoms
+                self._outputData[f"f(q,t)_coh_{pair_str}"].scaling_factor *= (
+                    bi * bj * sqrt_cij
+                )
+                self._outputData[f"s(q,f)_coh_{pair_str}"].scaling_factor *= (
+                    bi * bj * sqrt_cij
+                )
             else:
-                self._outputData[f"f(q,t)_coh_{pair_str}"] *= 2 * bi * bj * norm_natoms
-                self._outputData[f"s(q,f)_coh_{pair_str}"] *= 2 * bi * bj * norm_natoms
+                self._outputData[f"f(q,t)_coh_{pair_str}"].scaling_factor *= (
+                    2 * bi * bj * sqrt_cij
+                )
+                self._outputData[f"s(q,f)_coh_{pair_str}"].scaling_factor *= (
+                    2 * bi * bj * sqrt_cij
+                )
```

Comment:
I do think if we're rewriting this it might be slightly clearer to have:
```python
pre_fac = 2 if pair[0] == pair[1] else 1
self._outputData[f"f(q,t)_coh_{pair_str}"].scaling_factor *= pre_fac * bi * bj * sqrt_cij
self._outputData[f"s(q,f)_coh_{pair_str}"].scaling_factor *= pre_fac * bi * bj * sqrt_cij
```

Reply:
Yes this is clearer I've changed this now.